tool/ood-gen/lib/event.ml

```diff
@@ -0,0 +1,59 @@
+type location = { lat : float; long : float }
+[@@deriving of_yaml, show { with_path = false }]
+
+type metadata = {
+  title : string;
+  url : string;
+  textual_location : string;
+  location : location option;
+  start_date : string;
+  start_time : float option;
+  end_time : float option;
+  end_date : string option;
+}
+[@@deriving of_yaml, show { with_path = false }]
+
+type t = {
+  title : string;
+  url : string;
+  slug : string;
+  textual_location : string;
+  location : location option;
+  start_date : string;
+  start_time : float option;
+  end_time : float option;
+  end_date : string option;
+}
+[@@deriving
+  stable_record ~version:metadata ~remove:[ slug ], show { with_path = false }]
+
+let of_metadata m = of_metadata m ~slug:(Utils.slugify m.title)
+let decode s = Result.map of_metadata (metadata_of_yaml s)
```

Comment:
Check out `industrial_user.ml` for an example of a `decode` function that decodes from `.md` files :slightly_smiling_face:

Reply:
@sabine Thank you. I've done that and it's working now.